lib/Dialect/ESI/runtime/cpp/lib/backends/Trace.cpp

```diff
@@ -271,12 +275,63 @@ TraceAccelerator::requestChannelsFor(AppIDPath idPath,
   return impl->requestChannelsFor(idPath, bundleType);
 }
 
+class TraceHostMem : public HostMem {
+public:
+  TraceHostMem(TraceAccelerator::Impl &impl) : impl(impl) {}
+
+  struct TraceHostMemRegion : public HostMemRegion {
+    TraceHostMemRegion(std::size_t size, TraceAccelerator::Impl &impl)
+        : impl(impl) {
+      ptr = malloc(size);
+      this->size = size;
```

Comment:
As above, wrt. constructor visibility. Even more relevant for this one, since the trace `allocate` write is performed in `TraceHostMem::allocate` and not in the `TraceHostMemRegion` constructor.